examples/12. Advanced Action.md

```diff
@@ -0,0 +1,66 @@
+FORMAT: 1A
+
+# Advanced Action API
+An action in a resource in the API Blueprint can be considered as a state transition. This API example demonstrates that a resource can contain a state transition to another resource. This allows us to show that the resource `Task` can afford an action that is not limited to the resource `Task`.
+
+## API Blueprint
++ [Previous: Resource Model](11.%20Resource%20Model.md)
++ [This: Raw API Blueprint](https://raw.github.com/apiaryio/api-blueprint/master/examples/11.%20Advanced%20Action.md)
+
+# Task [/tasks/{id}]
+
++ Parameters
+    + id (string)
+
+## Retrieve Task [GET]
+
++ Response 200 (application/json)
+
+        {
+            "id": 123,
+            "name": "Go to gym",
+            "done": false,
+            "type": "task"
+        }
+
+## Update Task [PATCH]
+This is a state transition within the same resource
```

Comment:
Yeah. I have wrestled with how to articulate this. However, if you think
about it, everytime you exercise a state transition you cycle to a new
"state". The fact that on the diagram it is the "same" state category can
obscure that it is a new state. Even calling `self` is a new state on a
resource since the data may have changed since you originally received it.
I say there are infinite data states here, but only so many categories of
states where the possible affordances change (as driven by some change in
the data state). Anyhow, that is how I think about it.

On Thu, Mar 5, 2015 at 8:28 AM, Z <notifications@github.com> wrote:

> In examples/12. Advanced Action.md
> <https://github.com/apiaryio/api-blueprint/pull/159#discussion_r25877018>:
>
> > ++ Parameters
> > +    + id (string)
> > +
> > +## Retrieve Task [GET]
> > +
> > ++ Response 200 (application/json)
> > +
> > +        {
> > +            "id": 123,
> > +            "name": "Go to gym",
> > +            "done": false,
> > +            "type": "task"
> > +        }
> > +
> > +## Update Task [PATCH]
> > +This is a state transition within the same resource
>
> So all in all depends on what you consider to be a state. From FSM
> standpoint not every action changes the state and I believe this is the
> state we are talking about when we are talking about transition changing
> the state (of FSM)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apiaryio/api-blueprint/pull/159/files#r25877018>.
>